Extension/src/LanguageServer/extension.ts

```diff
@@ -484,14 +484,18 @@ async function onSwitchHeaderSource(): Promise<void> {
         }
     });
     const document: vscode.TextDocument = await vscode.workspace.openTextDocument(targetFileName);
+    const workbenchConfig = vscode.workspace.getConfiguration("workbench");
     let foundEditor: boolean = false;
-    // If the document is already visible in another column, open it there.
-    vscode.window.visibleTextEditors.forEach((editor, index, array) => {
-        if (editor.document === document && !foundEditor) {
-            foundEditor = true;
-            vscode.window.showTextDocument(document, editor.viewColumn);
-        }
-    });
+    if (workbenchConfig.get("editor.revealIfOpen")) {
+        // If the document is already visible in another column, open it there.
+        vscode.window.visibleTextEditors.forEach((editor, index, array) => {
```

Comment:
I know you didn't write this code, but if you wouldn't mind updating this line of the lambda to the more concise code below while you're in here, I'd appreciate it.  (totally optional though)
```
vscode.window.visibleTextEditors.forEach(editor => {
```
There's no need to declare `index` and `array`